### ISSUE TITLE
MariaDB Database Server - MYSQL_PWD environment variable - this should be off by default for MariaDB installations

### ISSUE THREAD

**Issue by shoulders** [webmin/webmin#2308]
| SYSTEM INFORMATION ||
|----------|----------|
|  OS type and version |  Ubuntu Linux 22.04.5              |
|  Usermin version     |  2.102                             |
|  Virtualmin version  |  7.20.2 Pro                        |
|  Theme version       |  21.20.7                           |
|  Apache version      |  2.4.52                            |
|  Package updates     |  46 package updates are available  |

## Background

I have installed virtualmin and have never changed the MariaDB configuration.

## The issue

I made a modification to `/etc/mysql/my.cnf` by adding the `sql_mode="ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION"` and when I save I got this warning

![image](https://github.com/user-attachments/assets/40fafd94-1f23-4d9e-996c-7d55d031faf3)

I do not understand this. However I went and checked the configuration and the option `Use MYSQL_PWD variable to pass password?` was set to on.This was the same for my live and dev site. Just to re-iterate I have never made any changes to the MariaDB configuration.

![image](https://github.com/user-attachments/assets/8b7f1cb0-fe9f-4baa-8c2a-02a484b36050)

## Proposed solution
Make this off for new installations of virtualmin as they now all use MariaDB

## Question

- Should I turn this off?
- Is this being used by MariaDB at all?

## Additional

- The warning message was displayed with no menu or header.
- The warning message keeps replacing any MariaDB settings page so even though I am being given 2 options I only have one choice to disable the `MYSQL_PWD environment variable` and in which cas this should be disabled by default?
- Is the line of settings I added doing something adverse to the configuration parsing?

**Comment by jcameron** [comment]
Seems like maybe that addition to `my.cnf` broke the `mysql` command in a different way.

Can you still use it to connect to the database from the shell?

**Comment by jcameron** [comment]
So webmin needs some way to pass the password to the `mysql` and `mysqldump` commands. The alternative to `MYSQL_PWD` is to add it on the command line which is even less secure! Doing it via the `my.cnf` file isn't usable in practice because the MySQL module might need to connect as different users depending on the Webmin user.

That said, I think the error in your OP is misleading and is hiding the real problem. I will look into it some more..

**Comment by jcameron** [comment]
So I tested this by setting `sql_mode` to `ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION` , but was still able to connect just fine with Webmin using the MYSQL_PWD variable.

**Comment by shoulders** [comment]
> So I tested this by setting sql_mode to ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION , but was still able to connect just fine with Webmin using the MYSQL_PWD variable.

- When Virtualmin is installed, I am not sure that `MYSQL_PWD` is even enabled and it is just a parsing error by Webmin for the GUI.
- I think I could of put anything in the config which would cause the `my.cf` validation to pop the warning up 
- this option is not supported for MariaDB and should be in the off position.

My system worked fine when I set the option for `MYSQL_PWD` to off in webmin GUI. I am not sure this variable was actually ever really on, if it was my Virtualmin server has been running for about a year with no problems.

The only reason this became an issue for me is when I got the error message on-screen after I edited the `my.cnf`

weired

**Comment by jcameron** [comment]
Yeah I think this is a bug that's causing some other error to be interpreted as an authentication failure. But it's hard to figure out why unless I can re-produce it..

**Comment by iliaross** [comment]
Reopened, as this was closed by an automated cleanup run. @shoulders, sorry for the noise, and thanks for the contribution. We definitely didn't mean any disrespect.

If it turns out the issue is already resolved for you, you're welcome to close it.
